src/lib/y2storage/autoinst_proposal.rb

```diff
@@ -0,0 +1,184 @@
+      propose_guided_devicegraph(devicegraph, drives)
+    end
+
+    # Finds a suitable devicegraph using the guided proposal approach
+    #
+    # If the :desired target fails, it will retry with the :min target.
+    #
+    # @param devicegraph [Devicegraph]       Starting point
+    # @param drives      [AutoinstDrivesMap] Devices map from an AutoYaST profile
+    # @return [Devicegraph] Proposed devicegraph using the guidede proposal approach
+    #
+    # @raise [Error] No suitable devicegraph was found
+    # @see proposed_guided_devicegraph
+    def propose_guided_devicegraph(devicegraph, drives)
+      guided_devicegraph_for_target(devicegraph, drives, :desired)
+    rescue Error
+      guided_devicegraph_for_target(devicegraph, drives, :min)
+    end
+
+    # Calculates list of planned devices
+    #
+    # If the list does not contain any partition, then it follows the same
+    # approach than the guided proposal
+    #
+    # @param devicegraph [Devicegraph]                 Starting point
+    # @param drives      [Proposal::AutoinstDrivesMap] Devices map from an AutoYaST profile
+    # @return [Array<Planned::Device>] Devices to add
+    def plan_devices(devicegraph, drives)
+      planner = Proposal::AutoinstDevicesPlanner.new(devicegraph)
+      planner.planned_devices(drives)
+    end
+
+    # Creates a devicegraph using the same approach than guided partitioning
+    #
+    # @param devicegraph [Devicegraph]       Starting point
+    # @param drives      [AutoinstDrivesMap] Devices map from an AutoYaST profile
+    # @param target      [Symbol] :desired means the sizes of the partitions should
+    #   be the ideal ones, :min for generating the smallest functional partitions
+    def guided_devicegraph_for_target(devicegraph, drives, target)
+      guided_settings = proposal_settings_for_disks(drives.disk_names)
+      guided_planner = Proposal::PlannedDevicesGenerator.new(guided_settings, devicegraph)
+      planned_devices = guided_planner.planned_devices(target)
+      create_devices(devicegraph, planned_devices, drives.disk_names)
+    end
+
+    # Creates planned devices on a given devicegraph
+    #
+    # @param devicegraph     [Devicegraph]            Starting point
+    # @param planned_devices [Array<Planned::Device>] Devices to add
+    # @return [Devicegraph]
+    def create_devices(devicegraph, planned_devices, disk_names)
+      devices_creator = Proposal::AutoinstDevicesCreator.new(devicegraph)
+      devices_creator.devicegraph(planned_devices, disk_names)
+    end
+
+    # Returns the product's proposal settings for a given set of disks
+    #
+    # @param disk_names [Array<String>] Disks names to consider
+    # @return [ProposalSettings] Proposal settings considering only the given disks
+    def proposal_settings_for_disks(disk_names)
+      settings = ProposalSettings.new_for_current_product
+      settings.candidate_devices = disk_names
```

Comment:
what happens if none is defined? can it get nil or []? BTW to clarity it would be also good to mention what is supported format, as I think in autoyast you can specify udev names like `/dev/disk/by-id/blabla`